CommunicationLayer/CommDeviceControl/UdpMessageForwarder.cpp

```diff
@@ -26,27 +26,39 @@
 #include <QByteArray>
 #include <QDebug>
 #include <QHostAddress>
+#include <QTextCodec>
+
 
 #include "InfrastructureLayer/Settings/I_Settings.h"
 #include "UdpMessageForwarder.h"
 
+namespace
+{
+    const QString EXCHANGE_NAME = QString("amq.direct");
+}
+
 UdpMessageForwarder::UdpMessageForwarder(I_Settings& settings)
-: ipAddress_(settings.ipAddress())
-, port_(settings.udpPort())
 {
-    socket_.bind(ipAddress_, port_);
+  QString queueName = settings.queueName();
+
+  routingKey = settings.routingKey();
+
+  channel = Channel::Create(settings.ipAddress().toStdString(), (int)settings.udpPort());
+
+  channel->DeclareQueue(queueName.toStdString(), false, true, false, false);
+
+  channel->BindQueue(queueName.toStdString(), EXCHANGE_NAME.toStdString(), routingKey.toStdString());
+
 }
 
 UdpMessageForwarder::~UdpMessageForwarder()
 {
+  // TODO What to do for destructor? Nothing?
```

Comment:
The only concern that I have is that deleting the queue might delete the messages? That would be a huge problem. Otherwise, we should be cleaning up when we should haha